src/hypothesis/searchstrategy/types.py

```diff
@@ -0,0 +1,357 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import re
+import functools
+import collections
+from inspect import isclass
+
+import hypothesis.strategies as st
+from hypothesis.errors import InvalidArgument
+from hypothesis.internal.compat import text_type, integer_types, \
+    getfullargspec
+
+_global_type_to_strategy_lookup = {}
+
+
+def get_type_hints(thing):
+    """Try to return any type hints for ``thing``."""
+    try:
+        import typing
+        return typing.get_type_hints(thing)
+    except TypeError:
+        # `thing` is not a module, class, method, or function
+        return {}
+    except (ImportError, AttributeError):  # pragma: no cover
+        # This is a fallback for Python <3.6, where get_type_hints may fail
```

Comment:
Ugh, right. Looks like it fails for things implemented in C which as a result don't have a `__code__` attribute as a result.